src/analyses/base.ml

```diff
@@ -138,6 +138,11 @@ struct
   let project ask p_opt cpa fundec =
     CPA.mapi (fun varinfo value -> project_val ask (attributes_varinfo varinfo fundec) p_opt value (is_privglob varinfo)) cpa
 
+  let has_offset = function
+    | `NoOffset -> false
+    | `Field _
+    | `Index _ -> true
+
```

Comment:
It might actually even be cleaner to not have this function at all and just use `o <> `NoOffset` whereever it's needed.

Reply:
Even better in this case, yes. I wasn't sure if one could do such comparisons in OCaml. Thanks for the tip!

Reply:
Applied change in https://github.com/goblint/analyzer/pull/1139/commits/cb811e1f6d6ff42c4f3f62f93d18c34c767fad75